File: 26. Remove Duplicates from Sorted Array.java
/**
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    public int removeDuplicates(int[] nums) {
        int res = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i - 1] != nums[i]) {
                nums[++res] = nums[i];
            }
        }
        return res + 1;
    }
}

File: Kotlin/26. Remove Duplicates from Sorted Array.kt
/**
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    fun removeDuplicates(nums: IntArray): Int {
        var i = 0
        for (j in 1 until nums.size) {
            if (nums[j] != nums[i])
                nums[++i] = nums[j]
        }
        return i + 1
    }
}